mlir/include/mlir/Dialect/OpenACC/Transforms/Passes.td

```diff
@@ -27,4 +27,40 @@ def LegalizeDataValuesInRegion : Pass<"openacc-legalize-data-values", "mlir::fun
   ];
 }
 
+def ACCImplicitData : Pass<"acc-implicit-data", "mlir::ModuleOp"> {
+  let summary = "Generate implicit data attributes for OpenACC compute constructs";
+  let description = [{
+    This pass implements the OpenACC specification for "Variables with
+    Implicitly Determined Data Attributes" (OpenACC 3.4 spec, section 2.6.2).
+
+    The pass automatically generates data clause operations for variables used
+    within OpenACC compute constructs (parallel, kernels, serial) that do not
+    already have explicit data clauses. The semantics follow these rules:
+
+    1. If there is a default(none) clause visible, no implicit data actions
+       apply.
+
+    2. An aggregate variable (arrays, derived types, etc.) will be treated as:
+       - In a present clause when default(present) is visible.
+       - In a copy clause otherwise.
+
+    3. A scalar variable will be treated as if it appears in:
+       - A copy clause if the compute construct is a kernels construct.
+       - A firstprivate clause otherwise (parallel, serial).
+  }];
+  let dependentDialects = ["mlir::acc::OpenACCDialect",
+      "mlir::memref::MemRefDialect",
+      "mlir::arith::ArithDialect"];
```

Comment:
I might be wrong but I did not see any creation of ops from these two dialects? If you do not create ops from these dialects, you don't need them in the dependent list.

Reply:
Oh ok that makes sense.